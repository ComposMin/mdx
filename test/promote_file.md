Mdx can also understand ocaml code blocks:

```ocaml file=promote.ml,part=toto
# let x = 3;;
val x : int = 3
# let y = 4;;
val y : int = 4
# print_int x;;
3
- : unit = ()
# print_int y;;
4
- : unit = ()
```

```ocaml file=promote.ml,part=zzz
```

```ocaml file=promote.ml,part=42
```

```ocaml file=promote.ml,part=
```

```ocaml file=promote.ml
```

```ocaml
# print_int x;;
3
- : unit = ()
```
