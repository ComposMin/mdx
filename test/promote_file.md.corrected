Mdx can also understand ocaml code blocks:






```ocaml
# print_int x;;
3
- : unit = ()
```
