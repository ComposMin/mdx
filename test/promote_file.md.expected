Mdx can also understand ocaml code blocks:

```ocaml file=promote.ml,part=toto
let x = 34
let f = 42.3
let s = "toto"
let f x u = u x

let () =
  print_int x;
  print_float f
;;
```

```ocaml file=promote.ml,part=zzz
let () =
  print_string s
;;
```

```ocaml file=promote.ml,part=42
let () =
  f x print_int

```

```ocaml file=promote.ml,part=
let () =
  ();
  ()
;;
```

```ocaml file=promote.ml

let () =
  ();
  ()
;;

let x = 34
let f = 42.3
let s = "toto"
let f x u = u x

let () =
  print_int x;
  print_float f
;;

let () =
  print_string s
;;

let () =
  f x print_int

```

```ocaml
# print_int x;;
3
- : unit = ()
```
